crates/plugins/src/manifest.rs

```diff
@@ -0,0 +1,206 @@
+use anyhow::{anyhow, Context, Result};
+use semver::{Version, VersionReq};
+use serde::{Deserialize, Serialize};
+
+/// Expected schema of a plugin manifest. Should match the latest Spin plugin
+/// manifest JSON schema:
+/// https://github.com/fermyon/spin-plugins/tree/main/json-schema
+#[derive(Serialize, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct PluginManifest {
+    /// Name of the plugin.
+    name: String,
+    /// Option description of the plugin.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    description: Option<String>,
+    /// Optional address to the homepage of the plugin producer.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    homepage: Option<String>,
+    /// Version of the plugin.
+    pub(crate) version: String,
+    /// Versions of Spin that the plugin is compatible with.
+    pub(crate) spin_compatibility: String,
+    /// License of the plugin.
+    pub(crate) license: String,
+    /// Points to source package[s] of the plugin..
+    pub(crate) packages: Vec<PluginPackage>,
+}
+
+impl PluginManifest {
+    pub fn name(&self) -> String {
+        self.name.to_lowercase()
+    }
+}
+
+/// Describes compatibility and location of a plugin source.
+#[derive(Serialize, Debug, Deserialize, PartialEq)]
+pub struct PluginPackage {
+    /// Compatible OS.
+    pub(crate) os: Os,
+    /// Compatible architecture.
+    pub(crate) arch: Architecture,
+    /// Address to fetch the plugin source tar file.
+    pub(crate) url: String,
+    /// Checksum to verify the plugin before installation.
+    pub(crate) sha256: String,
+}
+
+/// Describes the compatible OS of a plugin
+#[derive(Serialize, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub(crate) enum Os {
+    Linux,
+    Macos,
+    Windows,
+}
+
+impl Os {
+    // Maps manifest OS options to associated Rust OS strings
+    // https://doc.rust-lang.org/std/env/consts/constant.OS.html
+    pub(crate) fn rust_name(os: &Os) -> String {
+        match os {
+            Os::Linux => "linux".to_string(),
+            Os::Macos => "macos".to_string(),
+            Os::Windows => "windows".to_string(),
```

Comment:
Enums can have methods just like structs. Also, because these are static strings, you can return `&'static str` instead of `String`:

```suggestion
    pub(crate) fn rust_name(&self) -> &'static str {
        match os {
            Os::Linux => "linux",
            Os::Macos => "macos",
            Os::Windows => "windows",
```